src/script/script.h

```diff
@@ -571,16 +572,30 @@ class CScript : public CScriptBase
         if (b.empty())
             return nFound;
         iterator pc = begin();
+        iterator copied = begin();
+        size_t nShift = 0;
         opcodetype opcode;
         do
         {
             while (end() - pc >= (long)b.size() && memcmp(&pc[0], &b[0], b.size()) == 0)
             {
-                pc = erase(pc, pc + b.size());
+                // Lazy copy-in-place if there is a match:
+                if (nShift == 0) copied = pc;
```

Comment:
Why initialize copied instead of just eliminating the if/else and always copying? Because std::copy behavior is undefined if the first and last arguments point to the same element.